res/res_pjsip_rfc3326.c

```diff
@@ -46,16 +46,34 @@ static void rfc3326_use_reason_header(struct ast_sip_session *session, struct pj
 	for (; header;
 		header = pjsip_msg_find_hdr_by_name(rdata->msg_info.msg, &str_reason, header->next)) {
 		int cause_q850, cause_sip;
+		struct ast_control_pvt_cause_code *cause_code;
+		int data_size = sizeof(*cause_code);
+
 		ast_copy_pj_str(buf, &header->hvalue, sizeof(buf));
 		cause = ast_skip_blanks(buf);
-
 		cause_q850 = !strncasecmp(cause, "Q.850", 5);
 		cause_sip = !strncasecmp(cause, "SIP", 3);
 		if ((cause_q850 || cause_sip) && (cause = strstr(cause, "cause="))) {
 			int *code = cause_q850 ? &code_q850 : &code_sip;
 			if (sscanf(cause, "cause=%30d", code) != 1) {
 				*code = 0;
 			}
+
+			/* Build and send the tech-specific cause information */
+			/* size of the string making up the cause code is "SIP " + reason length */
+			data_size += 4 + strlen(cause) + 1;
+			cause_code = ast_alloca(data_size);
+			memset(cause_code, 0, data_size);
+			ast_copy_string(cause_code->chan_name, ast_channel_name(session->channel), AST_CHANNEL_NAME);
+			snprintf(cause_code->code, data_size, "SIP %s", cause);
```

Comment:
Do not understand that comment: what should be done and what is the meaning of "safe" in this case?

Reply:
Actually since you're using ast_copy_string and snprintf, it can be left out. Beside operations that don't check size such as sprintf we put /* Safe */ as a comment so that if someone submits issues from a checker we can quickly say that we've already previously looked and that the operation is actually safe.